crates/protocol/src/batch/traits.rs

```diff
@@ -23,3 +27,35 @@ pub trait BatchValidationProvider {
     /// Errors if no block is available for the given block number.
     async fn block_by_number(&mut self, number: u64) -> Result<OpBlock, Self::Error>;
 }
+
+#[cfg_attr(target_arch = "wasm32", async_trait::async_trait(?Send))]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait::async_trait)]
+impl BatchValidationProvider for alloy_provider::ReqwestProvider {
+    type Error = TransportError;
+
+    async fn l2_block_info_by_number(&mut self, number: u64) -> Result<L2BlockInfo, Self::Error> {
+        todo!()
+    }
+
+    async fn block_by_number(&mut self, number: u64) -> Result<OpBlock, Self::Error> {
+        // Get the block from the provider.
+        let block = self.get_block_by_number(number.into(), BlockTransactionsKind::Full).await?;
+        let block = block.unwrap(); // TODO: handle error
+
+        // Convert the transactions.
+        // TODO: how/what do we actually need to do here?
```

Comment:
@refcell Could you please advise the canonical approach to getting the `OpBlock` type from the results returned from `alloy::Provider` fns like `get_block_by_number()` (used above). Main issue here is converting from the vec of transactions returned by the provider into `OpTxEnvelope` enum. Couldn't find any relevant `from` impls.

Or if there is an alternative means to performing the requests here. I looked at how Kona uses `ReqwestProvider` and found that it did `provider.client().request("debug_* ...` style requests for headers etc. Unsure if that is a relevant approach here.

TY!

Reply:
Thanks @emhane for that. I have pushed up a Reqwest impl for the BatchValidationProvider trait. I am guessing that this impl should be able to return more than just deposit txns which would mean it depends on your PR you linked there as you say.

If the intention is to only support deposit txns, then I still don't know how to convert between the eth rpc tx types and a tx type that the user of the Reqwest impl would expect.

I have left TODOs on the relevant lines. There are also TODOs for 2 fields I lack in order to construct the L2BlockInfo, LMK if you have any insight on that too.

TY!